include/unifex/just_void_or_done.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/blocking.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/std_concepts.hpp>
+
+#include <exception>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _just_void_or_done {
+
+template <typename Receiver>
+struct _op {
+  struct type;
+};
+template <typename Receiver>
+using operation = typename _op<remove_cvref_t<Receiver>>::type;
+
+template <typename Receiver>
+struct _op<Receiver>::type {
+  bool is_void_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
```

Comment:
I think you might get a smaller object if you reorder these members.  As far as I understand, `[[no_unique_address]]` allows the compiler to put subsequent members in trailing padding so, given that `sizeof(bool)` and `alignof(bool)` are both one, `is_void_` should fit into `receiver_`'s trailing padding if there is any.
```suggestion
  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
  bool is_void_;
```